src/compiler/program.ts

```diff
@@ -748,7 +748,21 @@ namespace ts {
         }
 
         if (!tryReuseStructureFromOldProgram()) {
-            forEach(rootNames, name => processRootFile(name, /*isDefaultLib*/ false));
+            let programSize = 0;
+            for (const name of rootNames) {
+                const path = toPath(name, currentDirectory, getCanonicalFileName);
+                if (programSize <= maxProgramSize) {
+                    processRootFile(name, /*isDefaultLib*/ false);
+                    if (!hasTypeScriptFileExtension(name) && filesByName.get(path)) {
+                        programSize += filesByName.get(path).text.length;
+                    }
+                }
+                else {
+                    programDiagnostics.add(createCompilerDiagnostic(Diagnostics.Too_many_javascript_files_in_the_project_Consider_add_to_the_exclude_list_in_the_config_file));

```

Comment:
I thought we were going to indicate the path to the file being processed when the limit was exceeded (as this is likely in whatever folder you should have been excluding)?